### PR TITLE
docs(docs-main): show component dependencies

### PR DESCRIPTION
**Pull request by borisdiakur**
# Description

This PR adds dependency information to the component documentation. For that to work it includes am 11ty config tweak which transforms relative links to absolute ones. Also this PR fixes minor layout issues in the docs.

## Type of change

- [x] Documentation improvement

# How Has This Been Tested?

- [x] tested manually; ⚠️ I haven't tested the effects of the change in an environment with a different base path than `/` (on GitHub pages), but am confident it will work. However, please review carefully! ⚠️

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing tests pass locally with my changes

### CODE REVIEW

File: .eleventy.js [hunk 1]
```diff
@@ -58,10 +60,30 @@ module.exports = function (eleventyConfig) {
...
+        if (link.startsWith('./')) {
+          const splitted = env.page.url.split('/')
+          splitted.splice(splitted.length - 1, 0, link.substr(2))
+          return base + splitted.join('/')
+        }
+        if (link.startsWith('../')) {
+          const splitted = env.page.url.split('/')
+          splitted.splice(splitted.length - 2, 1, link.substr(3))
+          return base + splitted.join('/')
+        }
```
Comment (renet) [inline .eleventy.js:78]:
```suggestion
        const splitted = env.page.url.split('/')
        if (link.startsWith('./')) {
          splitted.push(link.substr(2))
          return base + splitted.join('/')
        }
        if (link.startsWith('../')) {
          splitted.splice(splitted.length - 2, 1, link.substr(3))
          return base + splitted.join('/')
        }
```

File: src/docs/components/docs-main/docs-main.css [hunk 2]
```diff
@@ -213,8 +215,12 @@
       border: 0;
       height: var(--ld-sp-1);
 
-      &:last-of-type + p {
+      &:last-of-type {
         display: none;
+
+        & + p {
+          display: none;
+        }
```
Comment (renet) [inline src/docs/components/docs-main/docs-main.css:223]: That's not necessary, is it?
```suggestion
```

File: src/docs/components/docs-main/docs-main.css [hunk 3]
```diff
@@ -232,16 +238,18 @@
...
-    #graph,
+    /*
     .mermaid,
-    .mermaid + hr {
+    .mermaid + hr,
+    [id^='dependencies'],
+    [id^='used-by'],
+    [id^='used-by'] + ul,
+    [id^='depends-on'],
+    [id^='depends-on'] + ul*/
```
Comment (renet) [inline src/docs/components/docs-main/docs-main.css:248]: Please remove comment.

**Review by renet — CHANGES_REQUESTED** [review]

File: src/docs/components/docs-main/docs-main.css (hunk 2 above)
Comment (borisdiakur) [reply src/docs/components/docs-main/docs-main.css:223]: It is, in the docs of the tabs component.

File: .eleventy.js (hunk 1 above)
Comment (borisdiakur) [reply .eleventy.js:78]: Keeping as is. the variable is only needed if the condition resolves to true.

Comment (renet) [reply .eleventy.js:78]:
```suggestion
        if (link.startsWith('./')) {
          const splitted = env.page.url.split('/')
          splitted.push(link.substr(2))
          return base + splitted.join('/')
        }
        if (link.startsWith('../')) {
          const splitted = env.page.url.split('/')
          splitted.splice(splitted.length - 2, 1, link.substr(3))
          return base + splitted.join('/')
        }
```
Then at least do this...

File: src/docs/components/docs-main/docs-main.css (hunk 2 above)
Comment (renet) [reply src/docs/components/docs-main/docs-main.css:223]: How can something inside a `display: none;`-container ever be visible?

Comment (renet) [reply src/docs/components/docs-main/docs-main.css:223]: Ok, read it wrong ^^"

File: .eleventy.js (hunk 1 above)
Comment (borisdiakur) [reply .eleventy.js:78]: This may result in links such as this one: `/components/?yolo#something/ld-button`

**Review by renet — APPROVED** [review]

**Pull request merged**